src/Auth/BasicAuthAuthenticator.php

```diff
@@ -6,13 +6,14 @@
 use SilverStripe\Control\HTTPResponse_Exception;
 use SilverStripe\ORM\ValidationException;
 use SilverStripe\Security\BasicAuth;
+use SilverStripe\Security\Member;
 
 /**
  * An authenticator using SilverStripe's BasicAuth
  */
 class BasicAuthAuthenticator implements AuthenticatorInterface
 {
-    public function authenticate(HTTPRequest $request)
+    public function authenticate(HTTPRequest $request): ?Member
     {
         try {
             return BasicAuth::requireLogin($request, 'Restricted resource');
```

Comment:
```suggestion
            return BasicAuth::requireLogin($request, 'Restricted resource') ?: null;
```

`BasicAuth::requireLogin()` can also return bool

https://github.com/silverstripe/silverstripe-framework/blob/4/src/Security/BasicAuth.php#L114